toolkit/primitives/domain/src/lib.rs

```diff
@@ -793,3 +794,29 @@ mod tests {
 		assert_eq!(current_decoded.0, vec![9; 64]);
 	}
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+pub struct StakePoolKeyHash(pub [u8; 28]); // hash28type
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+pub struct DelegatorAddressHash(pub [u8; 29]); // addr29type
```

Comment:
This is not domain type. The first byte is database indicator here.
I think that `DelegationKey` is rather an enum that is either 28 bytes of public staking key hash or a pair of 28 bytes of some key hash (I don't know exact type) and script hash itself.